Marlin/Marlin_main.cpp

```diff
@@ -3767,6 +3776,54 @@ inline void gcode_M206() {
 
 #endif // FWRETRACT
 
+#ifdef ENABLE_M211
+/**
+ * M211: Enable or disable software endstops
+ */
+inline void gcode_M211() {
+  // manage bit mask in enable_software_endstops
+  uint8_t smask = 0x00;
+
+  if (code_seen_str("X0")) smask |= 1;
+  if (code_seen_str("Y0")) smask |= 2;
+  if (code_seen_str("Z0")) smask |= 4;
+  if (code_seen_str("X1")) smask |= 16;
+  if (code_seen_str("Y1")) smask |= 32;
+  if (code_seen_str("Z1")) smask |= 64;
+
+
+  // Set mask
+  if (code_seen('S'))
+  {
+     if (code_value_long()==0) {
+        // clear mask
+        enabled_software_endstops &= ~smask;
+     }
+     else {
+        // set mask
+        enabled_software_endstops |=  smask;
+     }
+
+  }
+  else {
+    // Print state
+  SERIAL_PROTOCOLLN(MSG_M211_REPORT);
+  SERIAL_PROTOCOLPGM(MSG_X_MIN);
+  SERIAL_PROTOCOLLN(((enabled_software_endstops & 1) > 0?MSG_ENABLED:MSG_DISABLED));
+  SERIAL_PROTOCOLPGM(MSG_X_MAX);
+  SERIAL_PROTOCOLLN(((enabled_software_endstops & 16) > 0?MSG_ENABLED:MSG_DISABLED));
+  SERIAL_PROTOCOLPGM(MSG_Y_MIN);
+  SERIAL_PROTOCOLLN(((enabled_software_endstops & 2) > 0?MSG_ENABLED:MSG_DISABLED));
+  SERIAL_PROTOCOLPGM(MSG_Y_MAX);
+  SERIAL_PROTOCOLLN(((enabled_software_endstops & 32) > 0?MSG_ENABLED:MSG_DISABLED));
+  SERIAL_PROTOCOLPGM(MSG_Z_MIN);
+  SERIAL_PROTOCOLLN(((enabled_software_endstops & 4) > 0?MSG_ENABLED:MSG_DISABLED));
+  SERIAL_PROTOCOLPGM(MSG_Z_MAX);
+  SERIAL_PROTOCOLLN(((enabled_software_endstops & 64) > 0?MSG_ENABLED:MSG_DISABLED));
```

Comment:
There should probably be an enum for these bit masks to get rid of "magic numbers."
    enum EndStopBit { ESMinBitX = 0x01, ESMinBitY = 0x02, ESMinBitZ = 0x04, ESMaxBitX = 0x10, ESMaxBitY = 0x20, ESMaxBitZ = 0x40 };